avax-synapse-stablepool/contracts/Strategy.sol

```diff
@@ -0,0 +1,484 @@
+            _loss = debt.sub(currentValue);
+        }
+
+        uint256 toFree = _debtPayment.add(_profit);
+
+        if (toFree > wantBalance) {
+            toFree = toFree.sub(wantBalance);
+            (, uint256 withdrawalLoss) = withdrawSome(toFree);
+            // when we withdraw we can lose money in the withdrawl
+            if (withdrawalLoss < _profit) {
+                _profit = _profit.sub(withdrawalLoss);
+            } else {
+                _loss = _loss.add(withdrawalLoss.sub(_profit));
+                _profit = 0;
+            }
+
+            wantBalance = want.balanceOf(address(this));
+
+            if(wantBalance < _profit) {
+                _profit = wantBalance;
+                _debtPayment = 0;
+            } else if (wantBalance < _debtPayment.add(_profit)) {
+                _debtPayment = wantBalance.sub(_profit);
+            }
+        }
+
+        // we're done harvesting, so reset our trigger if we used it
+        forceHarvestTriggerOnce = false;
+    }
+
+
+    function withdrawSome(uint256 _amount) internal returns (uint256 _liquidatedAmount, uint256 _loss) {
+        uint256 wantBalanceBefore = want.balanceOf(address(this));
+
+        // let's take the amount we need if virtual price is real .
+        uint256 virtualPrice = virtualPriceToWant();
+        uint256 amountWeNeedFromVirtualPrice = _amount.mul(1e18).div(virtualPrice);
+
+        uint256 lpTokenBeforeBalance = emissionToken.balanceOf(address(this)); //should be zero but just incase...
+
+        uint256 pricePerFullShare = yvToken.pricePerShare();
+
+        uint256 amountFromVault = amountWeNeedFromVirtualPrice.mul(1e18).div(pricePerFullShare);
+
+        uint256 yBalance = yvToken.balanceOf(address(this));
+
+        if (amountFromVault > yBalance) {
+            amountFromVault = yBalance;
+            // this is not loss. so we amend amount
+
+            uint256 _amountOfCrv = amountFromVault.mul(pricePerFullShare).div(1e18);
+            _amount = _amountOfCrv.mul(virtualPrice).div(1e18);
+        }
+
+        if (amountFromVault > 0) {
+            yvToken.withdraw(amountFromVault);
+            if (withdrawProtection) {
+                //this tests that we liquidated all of the expected ytokens. Without it if we get back less then will mark it is loss
+                require(yBalance.sub(yvToken.balanceOf(address(this))) >= amountFromVault.sub(1), "YVAULTWITHDRAWFAILED");
+            }
+        }
+
+        uint256 toWithdraw = emissionToken.balanceOf(address(this)).sub(lpTokenBeforeBalance);
+
+        if (toWithdraw > 0) {
+            //if we have less than 18 decimals we need to lower the amount out
+            uint256 maxSlippage = toWithdraw.mul(DENOMINATOR.sub(slippageProtectionOut)).div(DENOMINATOR);
+            if(want_decimals < 18){
+                maxSlippage = maxSlippage.div(10 ** (uint256(uint8(18) - want_decimals)));
+            }
+
+            swapPool.removeLiquidityOneToken(toWithdraw, curveId, 0, now);
```

Comment:
Missing moment where it wraps the native token after withdrawing it.

Eg.
`if (isWantWETH()) {
                weth.deposit{value: address(this).balance}();
            }`